### PR TITLE
routing slip validation added

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/9121

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/internal_pay_service.py [hunk 1]
```diff
@@ -116,3 +115,27 @@ def complete_post_invoice(self, invoice: Invoice, invoice_reference: InvoiceRefe
...
+        # is rs doesnt exist , legacy routing slip flag should be on
+        if routing_slip is None:
+            if not current_app.config.get('ALLOW_LEGACY_ROUTING_SLIPS'):
+                raise BusinessException(Error.RS_DOESNT_EXIST)
+            # legacy routing slip which doesnt exist in the system.No validations
+            return
+
+        # check rs is active
+
+        if routing_slip.status in (
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/internal_pay_service.py:131]: May be we can check any status other than ACTIVE and LINKED ?

File: pay-api/src/pay_api/services/internal_pay_service.py [hunk 2]
```diff
@@ -116,3 +115,27 @@ def complete_post_invoice(self, invoice: Invoice, invoice_reference: InvoiceRefe
...
+                RoutingSlipStatus.BOUNCED.value, RoutingSlipStatus.NSF.value, RoutingSlipStatus.COMPLETE.value):
+            raise BusinessException(Error.RS_NOT_ACTIVE)
+
+        if routing_slip.parent:
+            error = f'This Routing Slip is linked, enter the parent Routing slip:{routing_slip.parent.number}'
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/internal_pay_service.py:136]: Routing Slip, s is not consistent. It's `S` and `s`.

**Review by sumesh-aot — APPROVED** [review]

**Comment by sumesh-aot** [comment]
Please check comments

File: pay-api/src/pay_api/services/internal_pay_service.py (hunk 1 above)
Comment (saravanpa-aot) [reply pay-api/src/pay_api/services/internal_pay_service.py]: done..thats a good idea

File: pay-api/src/pay_api/services/internal_pay_service.py (hunk 2 above)
Comment (saravanpa-aot) [reply pay-api/src/pay_api/services/internal_pay_service.py]: hm..changed to all small case s... copied from the ticket though

**Review by karthik-aot — APPROVED** [review]

**Pull request merged**